presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/JtsGeometryUtils.java

```diff
@@ -97,7 +113,53 @@ public static Geometry deserialize(Slice shape)
         if (geometries.size() == 1) {
             return geometries.get(0);
         }
-        return GEOMETRY_FACTORY.createGeometryCollection(geometries.toArray(new Geometry[0]));
+
+        Geometry res = GEOMETRY_FACTORY.createGeometryCollection(geometries.toArray(new Geometry[0]));
+        res.setSRID(spatialReferenceId);
+        return res;
+    }
+
+    /**
+     * Serialize JTS {@link Geometry} shape into an ESRI shape
+     */
+    public static Slice serialize(Geometry geometry) {
+        int spatialReferenceId = geometry.getSRID();
+        DynamicSliceOutput sliceOutput = new DynamicSliceOutput(100);
+        sliceOutput.appendInt(spatialReferenceId);
+
+        int availableGeometry = geometry.getNumGeometries();
+        for (int i = 0; i < availableGeometry; i++) {
+            Geometry unitGeometry = geometry.getGeometryN(i);
+            switch (JtsShapeType.valueOf(unitGeometry.getGeometryType())) {
+                case POINT:
+                    writePoint((Point) unitGeometry, sliceOutput);
+                    break;
+                case MULTI_LINESTRING:
```

Comment:
You need to handle all of the following types: Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon. Currently, LineString and MultiPolygon are missing.

> (GeometryCollection) unitGeometry and (GeometryCollection) unitGeometry

Shouldn't these be `(MultiLineString) unitGeometry` and `(Polygon) unitGeometry` ?

Reply:
In the `readGeometry` function, it seems it's only handling POINT, POLYLINE, POLYGON and MULTI_POINT
why are LineString and MultiPolygon relevant if `public static Geometry deserialize(Slice shape)` is not handling it?

Reply:
ESRI serialization format uses Polyline type for both LineString and MultiLineString. readPolyline returns LineString if Polyline has only one part and MultiLineString if there are multiple parts. Same for readPolygon.

Reply:
Oh I see! Makes a lot sense to me now. Sorry for missing that